security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -187,6 +190,60 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, proxyID, resourceName
 	return ns, nil
 }
 
+// DeleteK8sSecret deletes all entries that match secretName. This is called when a K8s secret
+// for ingress gateway is deleted.
+func (sc *SecretCache) DeleteK8sSecret(secretName string) {
+	sc.secrets.Range(func(k interface{}, v interface{}) bool {
+		key := k.(ConnKey)
+
+		if key.ResourceName == secretName {
+			sc.secrets.Delete(key)
+		}
+		return true
+	})
+}
+
+// UpdateK8sSecret updates all entries that match secretName. This is called when a K8s secret
+// for ingress gateway is updated.
+func (sc *SecretCache) UpdateK8sSecret(secretName string, ns model.SecretItem) {
+	secretMap := map[ConnKey]*model.SecretItem{}
+	wg := sync.WaitGroup{}
+	sc.secrets.Range(func(k interface{}, v interface{}) bool {
```

Comment:
noneed to iterate all items in sc.secrets, just update the one that updated and call the callback.